### PR TITLE
[batch] expose gcsfuse to user

### PR DESCRIPTION
**Pull request by jigold**
Stacked on #8960

CHANGELOG: Added support for gcsfuse when using the ServiceBackend.

### CODE REVIEW

File: hail/python/hailtop/batch/job.py [hunk 1]
```diff
@@ -521,6 +522,43 @@ def timeout(self, timeout):
...
+        bucket: :obj:`str`
+            Name of the google storage bucket to mount.
+        mount_point: :obj:`str`
+            The path for which the bucket should be mounted to in the Docker
+            container.
+
+        Returns
+        -------
+        :class:`.Job`
+            Same job object set with a timeout.
```
Comment (danking) [inline hail/python/hailtop/batch/job.py:553]: I think this needs to be updated

File: hail/python/hailtop/batch/job.py [hunk 2]
```diff
@@ -521,6 +522,43 @@ def timeout(self, timeout):
...
+        >>> j = b.new_job()
+        >>> (j.gcsfuse('my-bucket', '/my-bucket')
+        ...   .command(f'cat /my-bucket/my-file'))
+
+        Parameters
+        ----------
+        bucket: :obj:`str`
+            Name of the google storage bucket to mount.
+        mount_point: :obj:`str`
+            The path for which the bucket should be mounted to in the Docker
```
Comment (danking) [inline hail/python/hailtop/batch/job.py:547]: at which the bucket

File: hail/python/test/hailtop/batch/test_batch.py [hunk 3]
```diff
@@ -447,6 +448,19 @@ def test_verbose(self):
...
+    def test_gcsfuse(self):
+        b = self.batch()
+
+        head = b.new_job()
+        head.command(f'echo head > /{self.bucket_name}{self.gcs_output_path}/gcsfuse_test')
+        head.gcsfuse(self.bucket_name, f'/{self.bucket_name}')
+
+        tail = b.new_job()
+        tail.command(f'cat /{self.bucket_name}{self.gcs_output_path}/gcsfuse_test')
+        tail.gcsfuse(self.bucket_name, f'/{self.bucket_name}')
```
Comment (danking) [inline hail/python/test/hailtop/batch/test_batch.py:460]: I think this will race with head, right? Maybe use two batches?

**Comment by jigold** [comment]
Adding the WIP label. Will untag it once all instances are version 5 (standing worker is version 4).

**Comment by jigold** [comment]
I'm going to temporarily close this while I debug with dev deploy.

**Pull request closed without merging**